### ISSUE TITLE
Is SonarJs has a rule like eslint's no-undef?

### ISSUE THREAD

**Issue by shikelong** [SonarSource/sonar-javascript#621]
http://eslint.org/docs/rules/no-undef

Regards.

**Comment by shikelong** [comment]
Variables should be declared before they are used seem can't find this issue:

```javascript

someFunctionNotDefined();

```

```javascript

var a = 'asd';

export {
 b
}

```

**Comment by shikelong** [comment]
Non-existent variables should not be referenced can't find the issue

```js

var a = 'asd'

export {
  b
}
```

**Comment by inverno** [comment]
@shikelong those rules are targeting different cases, you are probably looking for "Variables should be declared explicitly", see : https://sonarcloud.io/organizations/inverno-github/rules#rule_key=javascript%3AS2703

**Comment by shikelong** [comment]
The rule "Variables should be declared explicitly" has been activate in my Quality Profile.
But It seem no work when I export a un declared variable

**Comment by inverno** [comment]
I thought you were targeting implicit declarations, not just general uses of non-existent variables. "Non-existent variables should not be referenced" should be catching that, and it does, just not for export statements.
So, the answer to your question is that "Non-existent variables..." is what you need, but it doesn't seem to be working for your specific example with exports. I'll create a new issue with label bug and reference this one from there.
